### PR TITLE
Output collections

### PR DESCRIPTION
**Pull request by Oddant1**
Adds cli support for output collections.

### CODE REVIEW

**Comment by Oddant1** [comment]
blocked by qiime2/qiime2#675

File: q2cli/click/option.py [hunk 1]
```diff
@@ -217,13 +284,57 @@ def type_cast_value(self, ctx, value):
...
+            raise ValueError('Keyed values cannot be mixed with unkeyed '
+                             'values.')
+
+        return keys, values
+
     def _check_length(self, value, ctx):
         import collections
 
+        # TODO: Ok seriously though figure out why value is in a tuple if it is
+        # a Collection
```
Comment (ebolyen) [inline q2cli/click/option.py:331]: https://github.com/pallets/click/blob/main/src/click/core.py#L2319-L2320

**Review by ebolyen — APPROVED** [review]
There's an edge-case in loading an ordered directory which can swallow the error, but otherwise this all looks good. Sorry it took so long!

File: q2cli/util.py [hunk 2]
```diff
@@ -361,7 +393,49 @@ def _load_input(fp, view=False):
...
+
+def _load_collection(fp):
+    import os
+    import warnings
+
+    artifacts = {}
+    order_fp = os.path.join(fp, '.order')
+
+    if os.path.exists(order_fp) and os.path.isfile(order_fp):
+        artifacts, error = _load_ordered_collection(fp, order_fp)
```
[the remark this answers is not in the document]
Comment (Oddant1) [reply q2cli/util.py:407]: Hmmm, not 100% related, but can't I just do isfile there? That also says no if it doesn't exist

File: q2cli/util.py [hunk 3]
```diff
@@ -361,7 +393,49 @@ def _load_input(fp, view=False):
...
+    else:
+        warnings.warn(f'The directory {fp} does not contain a .order file. '
+                      'The files will be read into the collection in the '
+                      'order the filesystem provides them in.')
```
[the remark this answers is not in the document]
Comment (Oddant1) [reply q2cli/util.py:411]: I should have a test for this

File: q2cli/commands.py [hunk 4]
```diff
@@ -372,18 +372,51 @@ def __call__(self, **kwargs):
             os.makedirs(output_dir)
 
         for result, output in zip(results, outputs):
+            # TODO: Having a collection output causes this to become a tuple
+            # for some reason. I don't understand why yet
```
[the remark this answers is not in the document]
Comment (Oddant1) [reply q2cli/commands.py:376]: Thank you for tracking that down

**Pull request merged**